pkg/controller/mutators/assign/assign_controller.go

```diff
@@ -193,6 +204,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, nil
 	}
 
+	// Since we aren't deleting the mutator, we are either adding or updating
+	ingestionStatus := ctrlmutators.MutatorStatusError
+	defer func() {
```

Comment:
Do we want metrics for only this case? In either case, this would be cleaner as its own method instead of inline. Deferred functions over a few lines long can be messy to read/reason about.

Reply:
Ah, my mistake.  This is a good example of why we need to DRY up this code!  I followed your advice and already did this on `assignmetadata_controller.go`, but forgot to do it here.

Will mirror what I did in the other controller.